File: Back-End-M/src/main/java/com/attendance/fpt/enums/ComplaintType.java
package com.attendance.fpt.enums;

import lombok.Getter;

@Getter
public enum ComplaintType {
    MISSED_CHECKIN("Quên Checkin"),
    MISSED_CHECKOUT("Quên Checkout"),
    WRONG_TIME("Sai Giờ"),
    WRONG_LOCATION("Sai Địa Điểm"),
    SYSTEM_ERROR("Lỗi Hệ Thống"),
    OTHER("Khác");

    private final String displayName;

    ComplaintType(String displayName) {
        this.displayName = displayName;
    }

}

File: Front-End/src/types/complaint.type.ts
export type ComplaintResponse = {
  id: number;
  reason: string;
  date: string;
  createdAt: string;
  responseDate?: string | null;
  responseNote?: string | null;
  responseBy?: string | null;
  employeeName: string;
  complaintType: string;
  status: ComplaintStatus;
  departmentName: string;
};

export interface ComplaintAddRequest {
  reason: string;
  date: Date;
  complaintType: string;
  requestChange: string;
}

export enum ComplaintStatus {
  PENDING = "PENDING",
  APPROVED = "APPROVED",
  REJECTED = "REJECTED",
  RECALLED = "RECALLED",
}

export enum ComplaintType {
  MISSED_CHECKIN = "Quên Checkin",
  MISSED_CHECKOUT = "Quên Checkout",
  WRONG_TIME = "Sai Giờ",
  WRONG_LOCATION = "Sai Địa Điểm",
  SYSTEM_ERROR = "Lỗi Hệ Thống",
  OTHER = "Khác",
}
